akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpHeaderParser.scala

```diff
@@ -544,8 +550,21 @@ private[http] object HttpHeaderParser {
                 case -1 ⇒ if (sb != null) sb.append(c).append(byteChar(input, ix + 1)).append(byteChar(input, ix + 2)).append(byteChar(input, ix + 3)) else null
                 case cc ⇒ appended2(cc)
               }
-            } else fail(s"Illegal character '${escape(c)}' in header value")
-          scanHeaderValue(hhp, input, start, limit)(nsb, nix)
+            } else {
+              mode match {
+                case ParserSettings.IllegalResponseHeaderValueProcessingMode.Error ⇒
+                  fail(s"Illegal character '${escape(c)}' in header value")
+                case ParserSettings.IllegalResponseHeaderValueProcessingMode.Warn ⇒
+                  // ignore the illegal character and log a warning message
+                  // TODO: log a warning message
```

Comment:
yep, that's what I want to do as I said in my first comment. My only concern is that it will change lots of methods' signature.